### PR TITLE
Improper resource shutdown or release

### PR DESCRIPTION
**Pull request by katydidnot**
Bug: T358130

[//]: # (Thank you for uploading a PR to the Wikipedia Library!)

## Description
Describe your changes in detail following the [commit message guidelines](https://www.mediawiki.org/wiki/Gerrit/Commit_message_guidelines)

Closing all files that are opened with the open() function when files are done being read. 

## Rationale
[//]: # (Why is this change required? What problem does it solve?)

> 
Closing files is especially important when accessing file fields in a loop over a large number of objects. If files are not manually closed after accessing them, the risk of running out of file descriptors may arise. This may lead to the following error OSError: [Errno 24] Too many open files

per Django's documentation on [Managing Files](https://docs.djangoproject.com/en/3.2/topics/files/)

## Phabricator Ticket
[//]: # (Link to the Phabricator ticket)
https://phabricator.wikimedia.org/T358130

## How Has This Been Tested?
[//]: # (- Did you add tests to your changes? Did you modify tests to accommodate your changes?)
[//]: # (- Can this change be tested manually? How?)
I ran all unit tests and ensured they passed as well as performed a smoke test locally. 

## Screenshots of your changes (if appropriate):
[//]: # (It can also be a GIF to prove that your changes are working)

## Types of changes
What types of changes does your code introduce? Add an `x` in all the boxes that apply:
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Minor change (fix a typo, add a translation tag, add section to README, etc.)

### CODE REVIEW

File: TWLight/resources/helpers.py [hunk 1]
```diff
@@ -186,7 +186,7 @@ def _read_translation_file(language_code: str, filename: str):
     if os.path.isfile(filepath):
         with open(filepath, "r") as translation_file:
             translation_dict = json.load(translation_file)
-
+            translation_file.close()
```
Comment (jsnshrmn) [inline TWLight/resources/helpers.py:189]: I think the files that we open in `with` blocks are closed automatically

File: syslog/bin/import_logs.py [hunk 2]
```diff
@@ -938,7 +938,7 @@ def _parse_args(self, option_parser, argv = None):
 
         if self.options.output:
             sys.stdout = sys.stderr = open(self.options.output, 'a+', 0)
-
+            self.options.output.close()
```
Comment (jsnshrmn) [inline syslog/bin/import_logs.py:941]: I think this needs to stay open for the lifetime of the script execution for it to append logs as expected.

**Review by jsnshrmn — COMMENTED** [review]
Thanks for your work! comments inline

File: TWLight/resources/helpers.py (hunk 1 above)
Comment (katydidnot) [reply TWLight/resources/helpers.py:189]: You are correct. I totally read the Django docs incorrectly. **Always read the second sentence**. 🤦‍♀️ 
Screenshot 2024-05-10 at 11 44 36 AM

File: syslog/bin/import_logs.py (hunk 2 above)
Comment (katydidnot) [reply syslog/bin/import_logs.py]: I moved the file closures to after main() method runs and before we exit 0.

File: syslog/bin/import_logs.py [hunk 3]
```diff
@@ -1115,6 +1113,7 @@ def _get_token_auth(self):
 
             filename = stdout
             credentials = open(filename, 'r').readline()
+            filename.close()
```
Comment (katydidnot) [inline syslog/bin/import_logs.py:1116]: Should this be closed here?

Comment (jsnshrmn) [reply syslog/bin/import_logs.py:1116]: Yep, though the risk is super low since this code isn't web-accessible

**Review by jsnshrmn — APPROVED** [review]
This looks good and doesn't impact functionality in any way. Thanks for the patch!

**Pull request merged**